### PR TITLE
[DeviceMesh] Fixed `from_group` when passing tensor `mesh`

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #137721
* __->__ #137713

This fixes https://github.com/pytorch/pytorch/issues/137676. (sorry for messing this up in the original PR 😓 )

cc @XilunWu @H-Huang @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: torch/distributed/device_mesh.py [hunk 1]
```diff
@@ -761,7 +761,11 @@ def from_group(
                 group_ranks = get_process_group_ranks(group)
                 if (
                     isinstance(mesh, torch.Tensor) and mesh.tolist() != group_ranks
-                ) or (mesh is not None and mesh != group_ranks):
+                ) or (
+                    mesh is not None
+                    and not isinstance(mesh, torch.Tensor)
```
Comment (awgu) [inline torch/distributed/device_mesh.py:766]: add this check

**Review by wz337 — APPROVED** [review]
Thanks for the fix!

**Pull request closed without merging**